build_one.sh

```diff
@@ -10,4 +10,20 @@ if [ ! -f "$dockerfile_path" ]; then
 fi
 
 image_name=$(basename "$1")
-docker build -t "$org_name/$image_name" -f "$dockerfile_path" .
+tag_name="$org_name/$image_name"
+curr_tag_version=$(docker inspect "$tag_name:latest" --format '{{ index .Config.Labels "version" }}' || echo "")
+
+docker build -t "$tag_name:latest" -f "$dockerfile_path" .
+
+# Add version tag if Dockerfile is labelled with a version.
+new_tag_version=$(docker inspect "$tag_name:latest" --format '{{ index .Config.Labels "version" }}')
+if [ -n "$new_tag_version" ]; then
+  docker tag "$tag_name:latest" "$tag_name:$new_tag_version"
+
+  # Only preserve the latest and the previous image.
+  if [ -n "$curr_tag_version" ] && [ "$new_tag_version" != "$curr_tag_version" ]; then
```

Comment:
The first time we execute this `$curr_tag_version` is going to be empty, so we will need to set the `previous` tag manually, right?